test/allow_SUITE.erl

```diff
@@ -0,0 +1,243 @@
+                , Options          % to send
+                , ReqBody          % to send
+                , Url              % to send
+                , ExpectedStatus  % test get this received
+                , ExpectedJsonPairs
+                ),
+
+
+    ok.
+
+
+%%%%%%%%%%%%%%%%%%%% Utilities
+
+%% utility to save putting this in each test
+send_recieve( ReqHeaders          % to send
+            , Options          % to send
+            , ReqBody          % to send
+            , Url              % to send
+            , ExpectedStatus  % test get this received
+            , ExpectedJsonPairs   % list of key/values in json
+            ) ->
+
+    MyPort = application:get_env(ocas, port, 8080),
+    %%lager:info("test_post:port= ~p", [MyPort]),
+    {ok, Conn} = shotgun:open("localhost", MyPort),
+    {ok, Response} = shotgun:post(Conn, Url, ReqHeaders, ReqBody, Options),
+    lager:info("response = ~p", [Response]),
+
+    %% get status code of response
+    #{ status_code := RespStatus } = Response,
+    %%lager:info("status = ~p", [RespStatus]),
+    %% test what received was what was expected
+    ExpectedStatus = RespStatus,
+
+    %% get headers
+    #{ headers := RespHeaders } = Response,
+    %%lager:info("headers = ~p", [RespHeaders]),
+
+    %% verify headers
+    { <<"server">>, <<"Cowboy">>} =  lists:keyfind( <<"server">>
+                                                  , 1
+                                                  , RespHeaders
+                                                  ),
+    { <<"date">>, _Date } =  lists:keyfind(<<"date">>, 1, RespHeaders),
+
+    %% check if has body
+    #{ body := RespBody } = Response,
+
+    %% check body is json
+    true = jsx:is_json(RespBody),
+
+    %% decode json into erlang map
+    JsonMap = jsx:decode( RespBody, [return_maps] ),
+
+    lager:info("ExpectedJsonPairs: ~p", [ExpectedJsonPairs]),
+    lager:info("JsonMap: ~p", [JsonMap]),
+
+    %% check key/value pairs are as expected
+    check_map(ExpectedJsonPairs, JsonMap),
+
+    %% return
+    ok.
+
+check_map( [], _JsonMap ) ->
+    %% done since list is empty
+    ok;
+
+check_map( [ {Key, ExpectedValue} | RestOfExpectedJsonPairs ], JsonMap ) ->
+    %% Grab  first item in list and verify
+    lager:info("Testing Key/ExpectedValue: ~p/~p", [Key, ExpectedValue]),
+
+    %% see if key is in JsonMap
+    %%    error formating added so can tell which key/value failed
+    case maps:is_key(Key, JsonMap) of
+        true ->
+            %% key exists so check value
+            case maps:get(Key, JsonMap) of
+                ExpectedValue ->
+                    %% key exists and value correct so go to next
+                    check_map( RestOfExpectedJsonPairs, JsonMap);
+                BadValue ->
+                    %% key exists but has wrong value so format test fail error
+                    ErrorText = list_to_binary(io_lib:format("~p is bad value for key ~p", [BadValue, Key])),
```

Comment:
Line 228 is too long:                     ErrorText = list_to_binary(io_lib:format("~p is bad value for key ~p", [BadValue, Key])),.